packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -72,11 +73,39 @@ import MAX_SIGNED_31_BIT_INT from './maxSigned31BitInt';
 let didWarnAboutBadClass;
 let didWarnAboutGetDerivedStateOnFunctionalComponent;
 let didWarnAboutStatelessRefs;
+let emptyContextThatWarnsOnAccess;
 
 if (__DEV__) {
   didWarnAboutBadClass = {};
   didWarnAboutGetDerivedStateOnFunctionalComponent = {};
   didWarnAboutStatelessRefs = {};
+
+  emptyContextThatWarnsOnAccess =
+    typeof Proxy === 'function'
+      ? new Proxy(emptyObject, {
+          get(target, name) {
+            const owner = ReactCurrentOwner.current;
+            let info = '';
+            if (owner !== null) {
+              const componentName = getComponentName(owner);
+              if (componentName != null) {
+                info = ` Check the definition of ${componentName}.\n`;
+              }
+            }
+            warning(
```

Comment:
Seems like we should de-dupe this warning?

Reply:
A global dedupe is probably fine. We don't really expect anyone to hit this.